tests/phpunit/tests/user/queryCache.php

```diff
@@ -0,0 +1,778 @@
+
+		$this->assertContains( $user_id, $found, 'Expected to find author in returned values.' );
+
+		$user = get_user_by( 'id', $user_id );
+		$user->remove_role( 'author' );
+
+		$q2 = new WP_User_Query(
+			array(
+				'role' => 'author',
+			)
+		);
+
+		$found = wp_list_pluck( $q2->get_results(), 'ID' );
+		$this->assertNotContains( $user_id, $found, 'Expected not to find author in returned values.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_set_user_role() {
+		$user_id = self::factory()->user->create( array( 'role' => 'author' ) );
+
+		$q1 = new WP_User_Query(
+			array(
+				'role' => 'author',
+			)
+		);
+
+		$found = wp_list_pluck( $q1->get_results(), 'ID' );
+
+		$this->assertContains( $user_id, $found, 'Expected to find author in returned values.' );
+
+		$user = get_user_by( 'id', $user_id );
+		$user->set_role( 'editor' );
+
+		$q2 = new WP_User_Query(
+			array(
+				'role' => 'author',
+			)
+		);
+
+		$found = wp_list_pluck( $q2->get_results(), 'ID' );
+		$this->assertNotContains( $user_id, $found, 'Expected not to find author in returned values.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_delete_user() {
+		$user_id = self::factory()->user->create();
+
+		$q1 = new WP_User_Query(
+			array(
+				'include' => array( $user_id ),
+			)
+		);
+
+		$found    = wp_list_pluck( $q1->get_results(), 'ID' );
+		$expected = array( $user_id );
+
+		$this->assertSameSets( $expected, $found, 'Find author in returned values' );
+
+		wp_delete_user( $user_id );
+
+		$q2 = new WP_User_Query(
+			array(
+				'include' => array( $user_id ),
+			)
+		);
+
+		$found = wp_list_pluck( $q2->get_results(), 'ID' );
+		$this->assertNotContains( $user_id, $found, 'Expected not to find author in returned values.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_do_not_cache() {
+		$user_id = self::factory()->user->create();
+
+		$args = array(
+			'fields'  => array(
+				'user_login',
+				'user_nicename',
+				'user_email',
+				'user_url',
+				'user_status',
+				'display_name',
+			),
+			'include' => array( $user_id ),
+		);
+
+		$q1       = new WP_User_Query( $args );
+		$found1   = $q1->get_results();
+		$callback = static function( $user ) {
+			return (array) $user;
+		};
+
+		$found1 = array_map( $callback, $found1 );
+
+		$queries_before = get_num_queries();
+		$q2             = new WP_User_Query( $args );
+		$found2         = $q2->get_results();
+		$found2         = array_map( $callback, $found2 );
+		$queries_after  = get_num_queries();
+
+		$this->assertSame( $queries_after - $queries_before, 2, 'Ensure that query is not cached' );
+		$this->assertSameSets( $found1, $found2, 'Expected results to match.', 'Ensure that to results match' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_update_user() {
+		$user_id = self::factory()->user->create();
```

Comment:
```suggestion
		$user_id = /* one of the shared fixtures created in the before class method */
```

Any reason this can't use an existing user? The user tables get reset at the conclusion of tests so it's non-destructive.

Reply:
Fixed https://github.com/WordPress/wordpress-develop/pull/3544/commits/5a5773cbc8b88fe5d0b9f1361c3485fe498e66da